lib/vmdb/plugins.rb

```diff
@@ -77,7 +77,14 @@ def provider_plugins
     def asset_paths
       @asset_paths ||= begin
         require_relative 'plugins/asset_path'
-        map { |engine| AssetPath.new(engine) if AssetPath.asset_path?(engine) }.compact
+        map do |engine|
+          source = bundler_specs_by_path[engine.root.to_s].source
+          is_path = Bundler::Source::Path === source
+          is_git = Bundler::Source::Git === source
+          development_gem = !is_path || is_git
```

Comment:
So, I think I see what you are trying to do, but I think we should go about it differently.  From what I can tell you are concerned with just not infecting gems when they live in the Bundler owned gems directory (whether real gem or a git based gem).

So, this can easily be handled by seeing if the plugins path is in bundler's install path.  This code should do:

```ruby
in_bundler_gems = engine.root.expand_path.to_s.start_with?(Bundler.install_path.expand_path.to_s)
AssetPath.new(engine, in_bundler_gems) if AssetPath.asset_path?(engine)
```

Then down below:

```ruby
        @node_modules = in_bundler_gems ? Pathname.new("/tmp/test/#{@namespace}/node_modules") : @path.join('node_modules')
```

Reply:
Aaah, yeah that might be better, definitely seems like this involves less poking inside bundler's ..insides :).

Updating.